code/game/machinery/outpost_electrolyzer.dm

```diff
@@ -0,0 +1,540 @@
+//Hydrogen pump stuff
+
+/obj/machinery/atmospherics/components/unary/hydrogen_pump
+	name = "hydrogen pump"
+	desc = "Lets you use merits to buy hydrogen."
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "hydrogen_pump"
+
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 50
+	active_power_usage = 1000
+
+	density = TRUE
+	max_integrity = 400
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 100, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 30)
+	layer = OBJ_LAYER
+	showpipe = TRUE
+	pipe_flags = PIPING_ONE_PER_TURF | PIPING_DEFAULT_LAYER_ONLY
+	var/merit
+
+/obj/machinery/atmospherics/components/unary/hydrogen_pump/examine(mob/user)
+	. = ..()
+	if(merit)
+		. += "<span class='notice'>[src] has [merit] merits, equaling [merit * MOLS_PER_MERIT] mols of hydrogen.</span>"
+	else
+		. += "<span class='notice'>[src] has no merits, get some from the electrolyzer or buy them to get hydrogen!</span>"
+	. += "<span class='notice'>[src] is currently [on ? "on" : "off"], and shuts off above [H2_PUMP_SHUTOFF_PRESSURE] kPa.</span>"
+	. += "<span class='notice'>[src] can be Alt-Clicked to eject merits.</span>"
+
+/obj/machinery/atmospherics/components/unary/hydrogen_pump/process_atmos()
+	update_parents()
+	var/datum/gas_mixture/air = airs[1] //hydrogen out
+	if(!on)
+		return
+	if(!merit || air.return_pressure() > H2_PUMP_SHUTOFF_PRESSURE)
+		on = FALSE
+		visible_message("<span class='danger'>[src] shuts off!</span>")
+		playsound(src, 'sound/machines/switch2.ogg', 10, FALSE)
+		return
+	var/meritused
+	if(merit >= MERITS_USED_PER_TICK)
+		merit -= MERITS_USED_PER_TICK
+		meritused = MERITS_USED_PER_TICK
+	else
+		meritused = merit
+		merit = 0
+		on = FALSE
+	air.adjust_moles(GAS_HYDROGEN, meritused * MOLS_PER_MERIT)
+	GLOB.hydrogen_stored -= meritused * MOLS_PER_MERIT
+	air.set_temperature(T20C) //hydrogen from adjust_mols takes the temp of the container, and if the container is empty it defaults to 0K. this works for now
+
+/obj/machinery/atmospherics/components/unary/hydrogen_pump/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/merit/bundle))
+		var/obj/item/merit/bundle/C = I
+		merit += C.value
+		to_chat(user, "<span class='notice'>You deposit [I], for a total of [merit] merits.</span>")
+		qdel(I)
+		return
+	return ..()
+
+/obj/machinery/atmospherics/components/unary/hydrogen_pump/attack_hand(mob/user)
+	if(..())
+		return
+	on = !on
+	playsound(src, 'sound/machines/switch3.ogg', 10, FALSE)
+	to_chat(user, "<span class='notice'>You toggle the pump [on ? "on" : "off"].</span>")
+	investigate_log("was turned [on ? "on" : "off"] by [key_name(user)]", INVESTIGATE_ATMOS)
+	update_appearance()
+
+/obj/machinery/atmospherics/components/unary/hydrogen_pump/AltClick(mob/user)
+	if(merit)
+		new /obj/item/merit/bundle(drop_location(), merit)
+		merit = FALSE
+		playsound(src, 'sound/items/poster_being_created.ogg', 10, FALSE)
+		to_chat(user, "<span class='notice'>You retrieve the hydrogen merits.</span>")
+	return ..()
```

Comment:
might be a little annoying to open the tile in the top-right if you're just checking to see if any merits are in the pump - maybe change it to give a message saying 'no merits are stored!' or something instead? Up to you though

Reply:
This still needs feedback.